test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/OpenIdConnectConfigurationTests.cs

```diff
@@ -221,5 +220,60 @@ public void RoundTripFromJson()
 
             TestUtilities.AssertFailIfErrors(context);
         }
+
+        [Fact]
+        public void EmptyCollectionSerialization()
+        {
+            var context = new CompareContext {Title = "EmptyCollectionSerialization"};
+            // Initialize an OpenIdConnectConfiguration object with all collections empty.
+            var oidcWithEmptyCollections = new OpenIdConnectConfiguration();
+            var oidcWithEmptyCollectionsJson = OpenIdConnectConfiguration.Write(oidcWithEmptyCollections);
+
+            IdentityComparer.AreEqual(oidcWithEmptyCollectionsJson, "{\"JsonWebKeySet\":null,\"SigningKeys\":[]}", context);
```

Comment:
I believe JsonWebKeySet and SigningKeys should not be there either (if the object only has default values in it, it should be represented as an empty json object).

Can we do the same thing for JsonWebKey and SecurityKey? We expose JsonWebKeys in the Keys endpoint.
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/master/src/Microsoft.IdentityModel.Tokens/SecurityKey.cs

Reply:
@javiercn  for SecurityKey we choose to ignore some properties see: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/src/Microsoft.IdentityModel.Tokens/SecurityKey.cs#L54

Did you want additional work?

Reply:
@javiercn  we also did some work in JsonWebKey have a look: 
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs#L272

Reply:
@brentschmaltz I didn't know about those changes

My only requirement is that empty arrays and null properties don't show up and that no additional property on the class shows up. So instead of

```
{
    "keys": [
        {
            "e": "AQAB",
            "key_ops": [],
            "kid": "27AE5DDFC1C57A8A9191DBE7A7C167282D89C7CC",
            "kty": "RSA",
            "n": "uTpbm5B6yNZlTyiEfQpb1lblMAinxrjGQqD6PBP3VxBObotHawTgySj1oMQuHtx8rNMdDv0ju11RKX6QgZvZFTbnaA0jN7WAfX-yBe1J7cmu2ikqXaDbeDmX1TP5WSltMwLHwbT6Ft964_KT_G2lGONVJ79xmxITHG3o80TX0EZOuPgVbz9BXHHSr-VXJWKQvayr6OwF-fHAYDGR3WZGg_sl2tF1mdojGZT-cxUb38xbqx8mVHhg3siJxrel32-NRuCdWJt5RukCnUraA9XCmROPOMvFaqWft1Lo50iSpal1KhpDMq1tu2m8jk7R3SSQFZ4jtpR1EW4hj0SuWCZASQ",
            "use": "sig",
            "x5c": [],
            "keySize": 2048,
            "hasPrivateKey": false,
            "keyId": null,
            "cryptoProviderFactory": {
                "customCryptoProvider": null
            }
        }
    ]
}
```

it should be

```
{
    "keys": [
        {
            "e": "AQAB",
            "kid": "27AE5DDFC1C57A8A9191DBE7A7C167282D89C7CC",
            "kty": "RSA",
            "n": "uTpbm5B6yNZlTyiEfQpb1lblMAinxrjGQqD6PBP3VxBObotHawTgySj1oMQuHtx8rNMdDv0ju11RKX6QgZvZFTbnaA0jN7WAfX-yBe1J7cmu2ikqXaDbeDmX1TP5WSltMwLHwbT6Ft964_KT_G2lGONVJ79xmxITHG3o80TX0EZOuPgVbz9BXHHSr-VXJWKQvayr6OwF-fHAYDGR3WZGg_sl2tF1mdojGZT-cxUb38xbqx8mVHhg3siJxrel32-NRuCdWJt5RukCnUraA9XCmROPOMvFaqWft1Lo50iSpal1KhpDMq1tu2m8jk7R3SSQFZ4jtpR1EW4hj0SuWCZASQ",
            "use": "sig"
        }
    ]
}
```


What version are these changes you mentioned on? Are there nightly packages for them?
@pranavkm what version of IdentityModel are we currently using?

Reply:
@javiercn these are in the nightlies: https://www.myget.org/gallery/azureadwebstacknightly

Reply:
AFAIK, we're referencing the versions available on nuget.org: https://www.nuget.org/packages/Microsoft.IdentityModel.Protocols.OpenIdConnect/2.1.4. We haven't picked up IdentityModel nightlies in a while now.

Reply:
@pranavkm we are working on releasing 5.2.0, we have published preview1 on nuget.org. Preview1 does not have this fix in.

Reply:
I'll speak to @javiercn and @kichalla on how to get this going. Some of our build infrastructure to pick up partner packages is paused right now for some MQ work and it might be a > week before we can get it rolling again.

Reply:
@pranavkm we will wait until you have validated 5.2.0 before GA.